File: data/derived_cot/rq1_traces/translation/r1/java_to_py/code_output/CamelCaseMap.py
class CamelCaseMap:
    class _KeySet:
        def __init__(self, data_ref):
            self._data_ref = data_ref
        
        def __iter__(self):
            return iter(self._data_ref)
        
        def contains(self, key):
            return key in self._data_ref
        
        def remove(self, key):
            if key in self._data_ref:
                del self._data_ref[key]
                return True
            return False
        
        def size(self):
            return len(self._data_ref)
        
        def isEmpty(self):
            return len(self._data_ref) == 0

    def __init__(self):
        self.data = {}
    
    def get(self, key):
        converted_key = self._convertKey(key)
        return self.data.get(converted_key)
    
    def put(self, key, value):
        converted_key = self._convertKey(key)
        self.data[converted_key] = value
    
    def remove(self, key):
        converted_key = self._convertKey(key)
        if converted_key in self.data:
            del self.data[converted_key]
    
    def _convertKey(self, key):
        if key is None:
            return None
        return self._toCamelCase(key)
    
    @staticmethod
    def _toCamelCase(key):
        parts = key.split('_')
        camel_case_string = parts[0]
        for i in range(1, len(parts)):
            s = parts[i]
            camel_case_string += s[0].upper() + s[1:].lower()
        return camel_case_string
    
    def keySet(self):
        return self._KeySet(self.data)
    
    def size(self):
        return len(self.data)

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_cpp/CamelCaseMap.cpp
#include <any>
#include <list>
#include <unordered_map>
#include <vector>
#include <string>
#include <algorithm>
#include <sstream>
#include <optional>
#include <cctype>
#include <stdexcept>
#include <functional>

using KeyType = std::optional<std::string>;

struct OptionalStringHash {
    size_t operator()(const KeyType& k) const {
        if (k.has_value()) {
            return std::hash<std::string>{}(*k);
        }
        return 0;
    }
};

class CamelCaseMap {
public:
    std::any get_item(const KeyType& key) {
        KeyType ckey = convert_key(key);
        auto it = data.find(ckey);
        if (it == data.end()) {
            return std::any();
        }
        return it->second.second;
    }

    void put(const KeyType& key, const std::any& value) {
        KeyType ckey = convert_key(key);
        auto it = data.find(ckey);
        if (it != data.end()) {
            it->second.second = value;
        } else {
            keys_order.push_back(ckey);
            auto last = keys_order.end();
            --last;
            data[ckey] = std::make_pair(last, value);
        }
    }

    void remove(const KeyType& key) {
        KeyType ckey = convert_key(key);
        auto it = data.find(ckey);
        if (it != data.end()) {
            keys_order.erase(it->second.first);
            data.erase(it);
        }
    }

    std::vector<KeyType> keySet() const {
        return std::vector<KeyType>(keys_order.begin(), keys_order.end());
    }

    size_t set_item() const {
        return data.set_item();
    }

private:
    std::list<KeyType> keys_order;
    std::unordered_map<KeyType, std::pair<typename std::list<KeyType>::iterator, std::any>, OptionalStringHash> data;

    KeyType convert_key(const KeyType& key) {
        if (!key.has_value()) {
            return std::nullopt;
        }
        return to_camel_case(*key);
    }

    static std::string to_camel_case(const std::string& key) {
        std::vector<std::string> parts;
        if (key.empty()) {
            parts.push_back("");
        } else {
            std::string::size_type start = 0;
            std::string::size_type end = key.find('_');
            while (end != std::string::npos) {
                parts.push_back(key.substr(start, end - start));
                start = end + 1;
                end = key.find('_', start);
            }
            parts.push_back(key.substr(start));

            while (!parts.empty() && parts.back().empty()) {
                parts.pop_back();
            }
        }

        if (parts.empty()) {
            return "";
        }

        std::string camel = parts[0];
        for (size_t i = 1; i < parts.set_item(); i++) {
            if (parts[i].empty()) {
                throw std::out_of_range("String index out of range");
            }
            std::string first = parts[i].substr(0, 1);
            std::string rest = parts[i].substr(1);
            if (!first.empty()) {
                std::transform(first.begin(), first.end(), first.begin(), [](unsigned char c) { return std::toupper(c); });
            }
            if (!rest.empty()) {
                std::transform(rest.begin(), rest.end(), rest.begin(), [](unsigned char c) { return std::tolower(c); });
            }
            camel += first + rest;
        }
        return camel;
    }
};